src/test/com/yetanalytics/datasim/xapi/profile/template/rule_test.clj

```diff
@@ -1,96 +1,661 @@
-      (is (nil? (s/explain-data ::xs/statement processed))))))
+  (testing "apply cmi5 Template:"
+    (doseq [{:keys [id rules]} cmi5-templates]
+      (testing (format "Template: %s" id)
+        (let [processed (r/apply-rules-gen cmi5-statement rules :seed gen-seed)]
+          (is (every? (partial r/follows-rule? processed)
+                      (map r/parse-rule rules)))
+          (is (nil? (s/explain-data ::xs/statement processed))))))))
+
+(comment
+  (r/apply-rules-gen
+   {"id" "fd41c918-b88b-4b20-a0a5-a4c32391aaa0",
+    "timestamp" "2015-11-18T12:17:00+00:00",
+    "actor"
+    {"objectType" "Agent",
+     "name" "Project Tin Can API",
+     "mbox" "mailto:user@example.com"},
+    "verb"
+    {"id" "http://example.com/xapi/verbs#sent-a-statement",
+     "display" {"en-US" "sent"}},
+    "object"
+    {"id" "http://example.com/xapi/activity/simplestatement",
+     "definition"
+     {"name" {"en-US" "simple statement"},
+      "description"
+      {"en-US"
+       "A simple Experience API statement. Note that the LRS \n\t\t\t\tdoes not need to have any prior information about the Actor (learner), the \n\t\t\t\tverb, or the Activity/object."}}}}
+   [{:location "$.object.definition.type"}]
+   #_[{:location "$.verb.display.en-US", :all ["Launched"]}
+    {:location "$.verb.display.zh-CN", :all ["展开"]}
+    {:location "$.verb.display.zh-CN", :none ["展开"]}]
+   #_[{:location "$.context.extensions['https://w3id.org/xapi/cmi5/context/extensions/launchmode']"
+       :presence "included"
+       :all      ["Review" "Normal" "Browse"]}]
+   :seed gen-seed) 
+
+   (r/follows-rule?
+    {"id" "fd41c918-b88b-4b20-a0a5-a4c32391aaa0",
+     "timestamp" "2015-11-18T12:17:00+00:00",
+     "actor"
+     {"objectType" "Agent",
+      "name" "Project Tin Can API",
+      "mbox" "mailto:user@example.com"},
+     "verb"
+     {"id" "http://example.com/xapi/verbs#sent-a-statement",
+      "display" {"en-US" "sent"}},
+     "object"
+     {"objectType" "Agent"
+      "name"       "Owen Overrider"
+      "mbox"       "mailto:owoverrider@example.com"}}
+    (r/parse-rule {:location "$.object.definition.type"
+                   :presence "included"}))
+
+   (r/apply-rules-gen
+    {"id" "fd41c918-b88b-4b20-a0a5-a4c32391aaa0",
+     "timestamp" "2015-11-18T12:17:00+00:00",
+     "actor"
+     {"objectType" "Agent",
+      "name" "Project Tin Can API",
+      "mbox" "mailto:user@example.com"},
+     "verb"
+     {"id" "http://example.com/xapi/verbs#sent-a-statement",
+      "display" {"en-US" "sent"}},
+     "object"
+     {"objectType" "Agent"
+      "name"       "Owen Overrider"
+      "mbox"       "mailto:owoverrider@example.com"}}
+    [{:location "$.object.definition.type"
+      :presence "included"}]
+    #_[{:location "$.context.extensions['https://w3id.org/xapi/cmi5/context/extensions/launchmode']"
+        :presence "included"
+        :all      ["Review" "Normal" "Browse"]}]
+    :seed gen-seed))
```

Comment:
I get a gen error on this, what's the significance of that? I'm assuming it's because of the Agent object, is this something we plan to handle?

Reply:
If it is known and targeted for fixing, maybe add a commented out test case with a TODO

Reply:
Ah, I see it is handled in the next branch.